patches/tModLoader/Terraria/ModLoader/LocalizationLoader.cs

```diff
@@ -180,6 +180,23 @@ private static List<(string key, string value)> LoadTranslations(Mod mod, GameCu
 
 				string translationFileContents = streamReader.ReadToEnd();
 
+				if (LanguageManager.Instance.watchedMod == mod.Name) {
+					string path = Path.Combine(ModCompile.ModSourcePath, mod.Name, translationFile.Name);
+					if (File.Exists(path)) {
+						try {
+							translationFileContents = File.ReadAllText(path);
+						}
+						catch (Exception) {
+							string newText = $"Failed to read {path}";
+							if (!Main.dedServ)
+								Main.NewText(newText);
+							else
+								Console.WriteLine(newText);
+							Logging.tML.Info(newText);
```

Comment:
The exception will trigger the `FirstChanceExceptionHandler` anyway, so there's probably no reason to log here as well.